src/components/global/ImageUpload.vue

```diff
@@ -126,27 +148,39 @@ export default {
             return true;
         },
 
-        validateFiles(files) {
+        async validateFiles(files) {
             let validFiles = [];
             let invalidFileFound = false;
             let largeFileFound = false;
             let duplicateFound = false;
 
-            files.forEach((file) => {
+            for (let file of files) {
+                if (file.name.endsWith('.heic')) {
+                    this.isLoading = true;
+                    try {
+                        file = await ImageUploadService.convertHEICtoJPEG(file);
+                    } catch (error) {
+                        this.showDangerToast(
+                            'Unable to upload file. Please try again',
+                        );
+                    } finally {
+                        this.isLoading = false;
+                    }
+                }
```

Comment:
This conversion can be after the validation check. So that if a file is not valid then we need to convert them.